### ISSUE TITLE
Issue when used with lower teeth

### ISSUE THREAD

**Issue by kTonpa** [Tai-Hsien/MeshSegNet#20]
Thank you for this Project.

It works really well on the upper teeth scan.
![Upper](https://user-images.githubusercontent.com/13318497/129803602-c03265d5-5282-4544-8d23-a232d745b755.png)

But when I tried to run on the lower jaw scan it fails completely.
![Lower](https://user-images.githubusercontent.com/13318497/129803609-f053437d-db84-495f-824e-f4f963e76de7.png)

Is this because of the model trained on few data, or am I doing something wrong?

Thanks.

**Comment by Tai-Hsien** [comment]
Hello,
I think you do everything right. When we trained the neural network, we considered upper and lower are different. A pre-trained for lower model is also available.

**Comment by yacbak** [comment]
> Hello,
> I think you do everything right. When we trained the neural network, we considered upper and lower are different. A pre-trained for lower model is also available at https://github.com/Tai-Hsien/MeshSegNet/blob/d77a84f901ef43dfbba807734af8be37cfe78f78/models/MeshSegNet_Man_15_classes_72samples_lr1e-2_best.tar.

Hello, You are an excellent researcher, I have learned a lot from your project, but I have some questions that I hope you can give me some advice. 
Have you trained two models to predict the upper and lower dental arch respectively? 
Is MeshSegNet_Man_15_classes_72samples_lr1e-2_best.tar the pre-training model of the lower dental arch that you provided? 
Is MeshSegNet_Max_15_classes_72samples_lr1e-2_best.tar the pre-training model of the upper dental arch that you provided? 
Does this mean that I need to get the upper dental arch training model from the upper dental arch data set and the lower dental arch training model from the lower dental arch data set? 
Thank you very much.

**Comment by kTonpa** [comment]
Hi @Tai-Hsien thanks for the help. it work very well on the lower teeth scan
![Lower_worked](https://user-images.githubusercontent.com/13318497/130138771-20f2ff0f-fc2b-451a-b721-25f1e8f3fafd.png)

I think it would be good if you update the README.md file to include the description about the two model "Lower and Upper teeth"
Btw do you know how to change the number of the color labels in paraview, so that assign correct color for each segmented teeth.

Thanks.

**Comment by Tai-Hsien** [comment]
Hello @kTonpa

Sorry for the late response. Yes, we treated upper and lower arch as different models. In Paraview, you can select "Interpret Values as Categories" in "Color Map Editor" (click "Edit" under Coloring on the left panel). Then you can decide the color for each category.

**Comment by Tai-Hsien** [comment]
> > Hello,
> > I think you do everything right. When we trained the neural network, we considered upper and lower are different. A pre-trained for lower model is also available at https://github.com/Tai-Hsien/MeshSegNet/blob/d77a84f901ef43dfbba807734af8be37cfe78f78/models/MeshSegNet_Man_15_classes_72samples_lr1e-2_best.tar.
> 
> Hello, You are an excellent researcher, I have learned a lot from your project, but I have some questions that I hope you can give me some advice.
> Have you trained two models to predict the upper and lower dental arch respectively?
> Is MeshSegNet_Man_15_classes_72samples_lr1e-2_best.tar the pre-training model of the lower dental arch that you provided?
> Is MeshSegNet_Max_15_classes_72samples_lr1e-2_best.tar the pre-training model of the upper dental arch that you provided?
> Does this mean that I need to get the upper dental arch training model from the upper dental arch data set and the lower dental arch training model from the lower dental arch data set?
> Thank you very much.

Yes, everything you said is right!

**Comment by kTonpa** [comment]
> Hello @kTonpa
> 
> Sorry for the late response. Yes, we treated upper and lower arch as different models. In Paraview, you can select "Interpret Values as Categories" in "Color Map Editor" (click "Edit" under Coloring on the left panel). Then you can decide the color for each category.


Yes, it worked well. here is the results:
Upper teeth scan
![Upper](https://user-images.githubusercontent.com/13318497/130656462-91b76afd-e736-4e3a-9da1-e28eca20a0f2.png)

Lower teeth scan
![Lower](https://user-images.githubusercontent.com/13318497/130656505-77a98162-d098-47f7-81c1-5829a2536287.png)

One more question please, Is the initial rotation of the input scan affect the segmentation result? because I noticed some minor changes in the segmentation result of 2 input scans made from the same scan rotated 90 degree apart.

Thank you very much again.

**Comment by Tai-Hsien** [comment]
> > Hello @kTonpa
> > Sorry for the late response. Yes, we treated upper and lower arch as different models. In Paraview, you can select "Interpret Values as Categories" in "Color Map Editor" (click "Edit" under Coloring on the left panel). Then you can decide the color for each category.
> 
> Yes, it worked well. here is the results:
> Upper teeth scan
> ![Upper](https://user-images.githubusercontent.com/13318497/130656462-91b76afd-e736-4e3a-9da1-e28eca20a0f2.png)
> 
> Lower teeth scan
> ![Lower](https://user-images.githubusercontent.com/13318497/130656505-77a98162-d098-47f7-81c1-5829a2536287.png)
> 
> One more question please, Is the initial rotation of the input scan affect the segmentation result? because I noticed some minor changes in the segmentation result of 2 input scans made from the same scan rotated 90 degree apart.
> 
> Thank you very much again.

Hello @kTonpa 

Glad that it works. You're right. A scan with different initial orientation will be considered as a different scan for the neural network, so the minor changes is reasonable. Ideally, the network should do the exactly same segmentation on the scans with different orientation, if the performance of network is general enough.

**Comment by kTonpa** [comment]
@Tai-Hsien thank you so much.
